plugins/unifying/fu-unifying-bootloader-texas.c

```diff
@@ -0,0 +1,231 @@
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-unifying-common.h"
+#include "fu-unifying-bootloader-texas.h"
+
+struct _FuUnifyingBootloaderTexas
+{
+	FuUnifyingBootloader	 parent_instance;
+};
+
+G_DEFINE_TYPE (FuUnifyingBootloaderTexas, fu_unifying_bootloader_texas, FU_TYPE_UNIFYING_BOOTLOADER)
+
+static gboolean
+fu_unifying_bootloader_texas_erase_all (FuUnifyingBootloader *self, GError **error)
+{
+	g_autoptr(FuUnifyingBootloaderRequest) req = fu_unifying_bootloader_request_new ();
+	req->cmd = FU_UNIFYING_BOOTLOADER_CMD_FLASH_RAM;
+	req->len = 0x01;	/* magic number */
+	req->data[0] = 0x00;	/* magic number */
+	if (!fu_unifying_bootloader_request (self, req, error)) {
+		g_prefix_error (error, "failed to erase all pages: ");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_unifying_bootloader_texas_compute_and_test_crc (FuUnifyingBootloader *self, GError **error)
+{
+	g_autoptr(FuUnifyingBootloaderRequest) req = fu_unifying_bootloader_request_new ();
+	req->cmd = FU_UNIFYING_BOOTLOADER_CMD_FLASH_RAM;
+	req->len = 0x01;	/* magic number */
+	req->data[0] = 0x03;	/* magic number */
+	if (!fu_unifying_bootloader_request (self, req, error)) {
+		g_prefix_error (error, "failed to compute and test CRC: ");
+		return FALSE;
+	}
+	if (req->cmd == FU_UNIFYING_BOOTLOADER_CMD_FLASH_RAM_WRONG_CRC) {
+		g_set_error_literal (error,
+				     G_IO_ERROR,
+				     G_IO_ERROR_FAILED,
+				     "CRC is incorrect");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_unifying_bootloader_texas_flash_ram_buffer (FuUnifyingBootloader *self, guint16 addr, GError **error)
+{
+	g_autoptr(FuUnifyingBootloaderRequest) req = fu_unifying_bootloader_request_new ();
+	req->cmd = FU_UNIFYING_BOOTLOADER_CMD_FLASH_RAM;
+	req->addr = addr;
+	req->len = 0x01;	/* magic number */
+	req->data[0] = 0x01;	/* magic number */
+	if (!fu_unifying_bootloader_request (self, req, error)) {
+		g_prefix_error (error, "failed to flash ram buffer @%04x: ", addr);
+		return FALSE;
+	}
+	if (req->cmd == FU_UNIFYING_BOOTLOADER_CMD_FLASH_RAM_INVALID_ADDR) {
+		g_set_error (error,
+			     G_IO_ERROR,
+			     G_IO_ERROR_FAILED,
+			     "failed to flash ram buffer @%04x: invalid flash page",
+			     addr);
+		return FALSE;
+	}
+	if (req->cmd == FU_UNIFYING_BOOTLOADER_CMD_FLASH_RAM_PAGE0_INVALID) {
+		g_set_error (error,
+			     G_IO_ERROR,
+			     G_IO_ERROR_FAILED,
+			     "failed to flash ram buffer @%04x: invalid App JMP vector",
+			     addr);
+		return FALSE;
+	}
+	if (req->cmd == FU_UNIFYING_BOOTLOADER_CMD_FLASH_RAM_INVALID_ORDER) {
+		g_set_error (error,
+			     G_IO_ERROR,
+			     G_IO_ERROR_FAILED,
+			     "failed to flash ram buffer @%04x: page flashed before page 0",
+			     addr);
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_unifying_bootloader_texas_clear_ram_buffer (FuUnifyingBootloader *self, guint16 addr, GError **error)
```

Comment:
This is only ever called with an address of 0 as I can tell, I'd say remove the `addr` parameter